plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx

```diff
@@ -202,6 +204,24 @@ export const Stepper = (stepperProps: StepperProps) => {
 
   const mergedUiSchema = merge({}, propUiSchema, currentStep?.uiSchema);
 
+  const mergedState = useMemo(() => {
+    if (!formState) {
+      return initialState;
+    }
+    const { [`step${activeStep}`]: activeState, ...historicalState } =
+      formState;
+    const chronologicalState = {
+      ...historicalState,
+      [`step${activeStep}`]: activeState,
+    };
+    return merge({}, ...Object.values(chronologicalState));
```

Comment:
I'm not really sure what this is doing?

```ts
const b = { asd: 'def', foo: 'bar' }
_.merge({}, ...Object.values(b)) // {0: 'b', 1: 'a', 2: 'r'}
```

Think it's a little hard to understand what's going on.

Reply:
It merges the steps data back to a merged state.
The reason to do this chronologically is to allow the active step to be the step to determine what end-up in the merged state.
So with step 1 active:

```typescript
const chronologicalState = { 'step-0': { name: 'foo', age: 40 }, 'step-1': { name: 'bar' } }
_.merge({}, ...Object.values(b)) // {name: 'bar', age: 40}
```
With step 0 active
```typescript
const chronologicalState = { 'step-1': { name: 'bar' }, 'step-0': { name: 'foo', age: 40 } }
_.merge({}, ...Object.values(b)) // {name: 'foo', age: 40}
```

Reply:
Ah right, can't we just grab the right one though out of the map depending on what the current step is? Don't really want to rely on insertion order into a dictionary if we can help it, feels risky.

Reply:
I don't really get the question. The determination of current step is not done in the merge. The merge is just merging the dictionary back to the current format of formData. Determining the current step and therefore the right data happens before where we create the chronologicalState, which is basically reordering the dictionary.

```typescript
const chronologicalState = {
      ...historicalState,
      [makeStepKey(activeStep)]: activeState,
    };
 ```
 
 I'm happy to improve, but I don't see much to improve in this part. I can add some specific unit tests for this logic to assure the functionality now and in the future.

Reply:
AH! Right sorry, now I understand what it's doing properly.

Yeah some tests for this might be nice just so that we know we don't break anything moving forward 🙏

Reply:
@benjdlambert I have implemented the extra test as discussed and implemented the suggested test case!